File: src/cpp_pkg/src/listener.cpp
#include "cpp_pkg/nodes.hpp"

Listener::Listener() : Node("listener") {
    subscription_ = create_subscription<std_msgs::msg::String>(
        "cpp_chatter", 10, [this](std_msgs::msg::String::SharedPtr message) {
            RCLCPP_INFO(get_logger(), "%s", message->data.c_str());
        });
}

File: src/python_pkg/python_pkg/listener.py
from rclpy.node import Node
from std_msgs.msg import String


class Listener(Node):
    def __init__(self):
        super().__init__("listener")
        self.subscription = self.create_subscription(
            String, "python_chatter", self.listen, 10
        )

    def listen(self, message):
        self.get_logger().info(message.data)
